internal/common/utils.go

```
@@ -0,0 +1,99 @@
+
+	errorResponse := map[string]interface{}{
+		"error": map[string]interface{}{
+			"code":    statusCode,
+			"message": message,
+		},
+	}
+
+	if err := json.NewEncoder(w).Encode(errorResponse); err != nil {
+		// If JSON encoding fails, fall back to plain text
+		w.Header().Set("Content-Type", "text/plain")
+		w.Write([]byte(fmt.Sprintf("Error: %s", message)))
```

golangci-lint (1.64.8)

Error return value of `w.Write` is not checked

(errcheck)